### PR TITLE
Make maintainer scripts dir relative to the package being built.

### PR DESCRIPTION
**Pull request by ximon18**
So that specifying a workspace package with -p works (see #151).

I haven't attempted to make this PR neat and tidy yet as I'm not sure if it's the right way to go or not. In particular the changes to the test code are a bit hacky at present. I'm curious to see what @kornelski and @qwandor think of it. With this change `cargo deb -p example` and `cd example ; cargo deb` both find the maintainer scripts, i.e. the issue reported on #151 is "fixed".

My main concern with this is to understand whether or not it constitutes a change in current behaviour or not, assuming @kornelski doesn't throw it right out that is ;-)

### CODE REVIEW

**Comment by qwandor** [comment]
Looks reasonable to me, though I haven't had a chance to try it yet.

**Comment by kornelski** [comment]
Looks good to me. I think the scripts were always meant to be crate-relative, so the change is a bugfix.

**Comment by ximon18** [comment]
I'll cleanup the PR and make it ready for final review then.

**Comment by ximon18** [comment]
@kornelski: please review, thanks.

**Comment by federico-terzi** [comment]
Hey @ximon18,

Thank you for your work! I've tested the PR with my project and I've noticed that the `maintainer-scripts` are only included if I specify the following (empty) section inside the Cargo.toml, as done in the [mijia-home](https://github.com/alsuren/mijia-homie/blob/master/mijia-homie/Cargo.toml#L63) project:

```
[package.metadata.deb.systemd-units]
```

Perhaps I'm missing something, but shouldn't the maintainer scripts be included even without systemd-specific settings?

File: src/control.rs [hunk 1]
```diff
@@ -370,19 +508,36 @@ mod tests {
...
+        // Archive more than once.
+        let mut out_ar = tar::Archive::new(&archive_bytes[..]);
+
+        let unreplaced_placeholders = out_ar
+            .entries()
+            .unwrap()
+            .map(Result::unwrap)
+            .map(|mut entry| {
+                let mut v = String::new();
+                entry.read_to_string(&mut v).unwrap();
```
Comment (kornelski) [inline src/control.rs:536]: It doesn't matter for tests, but FYI there's `std::fs::read_to_string`.

**Comment by ximon18** [comment]
Hi @federico-terzi,

Without it present maintainer scripts are still packaged into the DEB archive but not systemd unit specific processing of those scripts occurs, i.e. no service or other systemd specific files are detected and no code blocks are injected into the maintainer scripts to control the systemd unit during package installation and uninstallation.

Note that you can also include the table in compact form, e.g.:

```
systemd-units = { enable = false }
```

The reason that the systemd specific behaviour is not activated by default can be traced back to [this comment on the original PR](https://github.com/mmstick/cargo-deb/pull/135#issuecomment-666276027):

> I considered making this new behaviour the default, i.e. that cargo deb would be (ed: smart) enough to detect any systemd unit files in a debian subdirectory if present and to generate or augment maintainer scripts in debian and it would all just work. However, though perhaps this is not very likely, if someone had a debian directory containing systemd units that they did NOT want to be installed with the package (perhaps because the directory is shared between multiple packages or projects?) this new functionality would cause the package to be built differently than prior to this PR.

File: src/control.rs (hunk 1 above)
Comment (ximon18) [reply src/control.rs:536]: I thought I was forgetting something, but it was late and a quick search didn't find that, thanks!

**Comment by kornelski** [comment]
Thank you for the fix and the tests

File: src/control.rs (hunk 1 above)
Comment (ximon18) [reply src/control.rs:536]: Hmm, `std::fs::read_to_string()` only works for something `Path` like, but in this case we have an `Entry` from the `tar` crate. There's also [`std::io::read_to_string()`](https://doc.rust-lang.org/std/io/fn.read_to_string.html#) but that's only in nightly Rust.

Comment (kornelski) [reply src/control.rs:536]: Oh, you're right. I haven't noticed it's an entry.

**Comment by ximon18** [comment]
> Perhaps I'm missing something, but shouldn't the maintainer scripts be included even without systemd-specific settings?

@federico-terzi and @kornelski:  An example of why the systemd unit handling behavior isn't enabled automatically is regarding the rules for handling not found script files.

When the maintainer-scripts directory contains a service or other systemd supported file, the emulated Debian packaging logic requires that **not found** postinst, prerm or postrm files be generated.

If a user of cargo-deb had decided to control the systemd related behavior themselves with different logic than the Debian default, or had provided their own implementation prior to support for it existing in cargo-deb, the automatic addition of extra logic to their package when regenerating it using the latest cargo-deb could cause unwanted or unexpected changes in the behavior of their package at (un)install time.

If the cargo-deb owners (now just @kornelski ?) were to conclude that this is such a rare corner case and the impact on most users of being backward compatible is undesirable, we could of course decide to potentially break backward compatibility and diverge from the dh_installsystemd "official" Debian behavior by enabling the systemd related behavior by default, thereby making it easier for the majority use case, but I didn't and don't feel able to decide that on my own.

If that were to be done it should perhaps be mentioned in the release notes... except cargo-deb doesn't seem to have any...?

**Comment by federico-terzi** [comment]
@ximon18 Thank you for the detailed explanation!

**Pull request merged**